library/src/main/java/fr/tvbarthel/intentshare/IntentShare.java

```diff
@@ -171,6 +179,23 @@ public IntentShare iconLoader(@NonNull IconLoader iconLoader) {
     }
 
     /**
+     * Provide a custom {@link java.util.Comparator} in order to sort the {@link TargetActivity}
+     * displayed to the user.
+     * <p/>
+     * By default, TargetActivities are sorted by the recentness of their selection. If it's the
+     * behaviour that you are looking for, don't use this method and let the default comparator
+     * bring the magic.
+     *
+     * @param comparatorProvider comparator used to sort the TargetActivities displayed to the user.
+     *                           Will override the default sorting by recentness.
+     * @return current {@link IntentShare} for method chaining.
+     */
+    public IntentShare comparatorProvider(@NonNull TargetActivityComparatorProvider comparatorProvider) {
+        this.comparatorProvider = comparatorProvider;
+        return this;
+    }
```

Comment:
I think that the ```@NonNull``` annotation should not be used here.

According to the documentation the ```@NonNull``` annotation ```denotes that a parameter, field or method return value can never be null.``` 

From my understanding of the ```comparatorProvider``` methods, you do not control whether or not the comparatorProvider is null. If you do not want to support null parameter you should say so in the documentation and throw an NPE if comparatorProvider happens to be null.